### PR TITLE
chore: migration material UI

### PR DESCRIPTION
**Pull request by 0xtiti**
## Checklist before requesting a review

- [X] I have conducted a self-review of my code.
- [ ] If it is a core feature, I have included comprehensive tests.
- [ ] Do we need to implement analytics?
- [ ] Will this be part of a product update? If yes, please write one sentence about this update.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -27,6 +27,13 @@
     "src/**/*.{js,jsx,ts,tsx,css,scss,md,html}": "prettier --write --ignore-unknown"
   },
   "dependencies": {
+    "@emotion/cache": "11.11.0",
+    "@emotion/react": "11.11.3",
+    "@emotion/server": "11.11.0",
+    "@emotion/styled": "11.11.0",
+    "@mui/icons-material": "5.15.6",
+    "@mui/material": "^5.15.14",
```
Comment (0xArdy) [inline package.json]: please remove carets

File: src/providers/Web3ModalProvider.tsx [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+);
+
+const config = createConfig({
+  chains: [localhost],
+  transports: {
+    [localhost.id]: http(),
+  },
+  batch: { multicall: true },
+  connectors,
+});
```
Comment (0xArdy) [inline src/providers/Web3ModalProvider.tsx]: i run it locally and i cant connect to the app.

Comment (0xArdy) [inline src/providers/Web3ModalProvider.tsx]: We need to enable ssr in the wagmi config https://wagmi.sh/react/guides/ssr#ssr

File: src/providers/Web3ModalProvider.tsx [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+    },
+  ],
+  {
+    appName: 'Web3 React boilerplate',
+    projectId: PROJECT_ID,
+  },
+);
+
+const config = createConfig({
+  chains: [localhost],
```
Comment (0xArdy) [inline src/providers/Web3ModalProvider.tsx]: would be nice to have a testnet chain too, like Sepolia

File: src/providers/Web3ModalProvider.tsx [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+
+const { PROJECT_ID } = getConfig();
+
+const getWallets = () => {
+  if (PROJECT_ID) {
+    return [injectedWallet, rainbowWallet, walletConnectWallet];
+  } else {
+    return [injectedWallet];
+  }
+};
```
Comment (0xArdy) [inline src/providers/Web3ModalProvider.tsx]: awesome! now the app doesn't break when it doesn't find a project id :+1:

File: src/providers/ThemeProvider.tsx [hunk 5]
```diff
@@ -0,0 +1,54 @@
+import { ThemeProvider as MuiThemeProvider } from '@mui/material/styles';
+import { createContext, useEffect, useMemo, useState } from 'react';
```
Comment (0xArdy) [inline src/providers/ThemeProvider.tsx:1]: react imports should be on top

File: src/providers/ThemeProvider.tsx [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+import { getMuiThemeConfig } from '~/components/Theme/muiThemeConfig';
```
Comment (0xArdy) [inline src/providers/ThemeProvider.tsx]: i think you can import it as 
```ts
import { getMuiThemeConfig } from '~/components';
```

File: src/pages/layout.tsx [hunk 7]
```diff
@@ -1,18 +1,73 @@
...
+    textAlign: 'start',
+    fontSize: '1.6rem',
+    padding: '1rem 0.8rem 1rem',
+
+    p: {
+      padding: '1rem 0',
+      margin: 0,
+    },
+  };
+});
```
Comment (0xArdy) [inline src/pages/layout.tsx]: there is missing the styles to display it when it is a mobile view

File: src/pages/layout.tsx [hunk 8]
```diff
@@ -1,18 +1,73 @@
...
+        <NoScriptMessage>
+          <p>This website requires JavaScript to function properly.</p>
+        </NoScriptMessage>
+
+        {/* TODO: remove when responsive is done */}
+        <ResponsiveDisclaimer>
+          <p>
+            This website is not yet optimized for mobile devices. Please use a desktop browser for the best experience.
+          </p>
+        </ResponsiveDisclaimer>
```
Comment (0xArdy) [inline src/pages/layout.tsx:18]: nice touch :+1:

**Review by 0xArdy — CHANGES_REQUESTED** [review]

File: src/providers/Web3ModalProvider.tsx (hunk 2 above)
Comment (0xtiti) [reply src/providers/Web3ModalProvider.tsx]: 1862fff

Comment (0xtiti) [reply src/providers/Web3ModalProvider.tsx]: 1862fff

File: src/providers/Web3ModalProvider.tsx (hunk 3 above)
Comment (0xtiti) [reply src/providers/Web3ModalProvider.tsx]: 1862fff

File: src/providers/ThemeProvider.tsx (hunk 6 above)
Comment (0xtiti) [reply src/providers/ThemeProvider.tsx]: 1862fff

File: src/pages/layout.tsx (hunk 7 above)
Comment (0xtiti) [reply src/pages/layout.tsx]: 1862fff

File: package.json (hunk 1 above)
Comment (0xtiti) [reply package.json]: https://github.com/defi-wonderland/web3-react-boilerplate/commit/97036e6d0fd5d169e9b76518deac6aa18c861506

File: src/providers/ThemeProvider.tsx (hunk 5 above)
Comment (0xtiti) [reply src/providers/ThemeProvider.tsx:1]: https://github.com/defi-wonderland/web3-react-boilerplate/commit/97036e6d0fd5d169e9b76518deac6aa18c861506

File: src/providers/index.tsx [hunk 9]
```diff
@@ -0,0 +1,20 @@
+'use client';
```
Comment (0xArdy) [inline src/providers/index.tsx]: why? i think we don't need it here

File: src/providers/Web3ModalProvider.tsx [hunk 10]
```diff
@@ -0,0 +1,27 @@
+'use client';
```
Comment (0xArdy) [inline src/providers/Web3ModalProvider.tsx]: idem

File: src/providers/index.tsx (hunk 9 above)
Comment (0xtiti) [reply src/providers/index.tsx]: 56ec064

File: src/providers/Web3ModalProvider.tsx (hunk 10 above)
Comment (0xtiti) [reply src/providers/Web3ModalProvider.tsx]: 56ec064

**Review by 0xArdy — APPROVED** [review]
lgtm!

**Pull request merged**